python/pyspark/sql/pandas/serializers.py

```diff
@@ -155,12 +157,18 @@ def _create_batch(self, series):
         from pandas.api.types import is_categorical_dtype
         # Make input conform to [(series1, type1), (series2, type2), ...]
         if not isinstance(series, (list, tuple)) or \
-                (len(series) == 2 and isinstance(series[1], pa.DataType)):
+                (len(series) == 2 and isinstance(series[1], (pa.DataType, DataType))):
```

Comment:
I feel the current logic to handle arrow/spark types together makes _create_batch more complicated. We cannot pull out the logic to handle `UserDefinedType`s from this method?

Reply:
refactored